bot/handlers/not_sending_videos/save_clip_handler.py

```diff
@@ -35,206 +35,136 @@
-        segment = last_clip_info["segment"]
-        episode_info_data = segment.get("episode_info")
-
-        if isinstance(episode_info_data, EpisodeInfo):
-            episode_info_obj = episode_info_data
-        else:
-            episode_info_obj = EpisodeInfo(
-                season=episode_info_data["season"],
-                episode_number=episode_info_data["episode_number"],
-            )
-
-        return SegmentInfo(
-            video_path=last_clip_info["video_path"],
-            start=last_clip_info["start"],
-            end=last_clip_info["end"],
-            episode_info=episode_info_obj,
-            text=segment.get("text"),
-            id=segment.get("id"),
-            author=segment.get("author"),
-            comment=segment.get("comment"),
-            tags=segment.get("tags"),
-            location=segment.get("location"),
-            actors=segment.get("actors"),
+        return output_filename.replace(" ", "_"), segment_dict.get("start"), segment_dict.get(
+            "end",
+        ), False, season, episode_number
+
+    async def __handle_selected_clip(
+        self, last_clip: LastClip, segment_dict: Dict[str, any], output_filename: str,
+        season: Optional[int], episode_number: Optional[int],
+    ) -> Tuple[
+        str, float, float, bool, Optional[int], Optional[int],
+    ]:
+        await ClipsExtractor.extract_clip(
+            segment_dict.get("video_path"), last_clip.adjusted_start_time, last_clip.adjusted_end_time, output_filename,
+            self._logger,
         )
-
-    async def __prepare_clip_file(self, segment_info: SegmentInfo) -> Tuple[str, int, int, bool, Optional[int], Optional[int]]:
-        start_time = segment_info.start - settings.EXTEND_BEFORE
-        end_time = segment_info.end + settings.EXTEND_AFTER
-        is_compilation = False
-        season = None
-        episode_number = None
-
-        if segment_info.compiled_clip is not None:
-            output_filename = self.__write_clip_to_file(segment_info.compiled_clip)
-            is_compilation = True
-        else:
-            clip_path = segment_info.video_path
-            season = segment_info.episode_info.season
-            episode_number = segment_info.episode_info.episode_number
-            with tempfile.NamedTemporaryFile(delete=False, delete_on_close=False, suffix=".mp4") as tmp_file:
-                output_filename = tmp_file.name
-            await ClipsExtractor.extract_clip(clip_path, start_time, end_time, output_filename, self._logger)
-
-        return output_filename, start_time, end_time, is_compilation, season, episode_number
+        return output_filename.replace(
+            " ",
+            "_",
+        ), last_clip.adjusted_start_time, last_clip.adjusted_end_time, False, season, episode_number
+
+    async def __handle_single_clip(
+        self, last_clip: LastClip, segment_dict: Dict[str, any], output_filename: str,
```

Comment:
czy `Dict[str, any]` to nie jest po prostu json? chyba raczej tak to nazywaliśmy, przejrzyj sobie projekt i ujednolić

Reply:
z segment_dict zrobiłem segment_json zakładam że o to ci chodziło

Reply:
1) nazwa zmiennej nie powinna zawierać typu, bo po to jest typ xD po prostu segment
2) chodziło mi o używanie `segment: json` jeżeli faktycznie masz jsona, przygłup za 100zł miesięcznie ma nieaktualnego pytonga ewidentnie bo też mi wymyślał z str, any zamiast jsona jak człowiek